src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -96,20 +110,6 @@ public function createView(ChoiceListInterface $list, $preferredChoices = null,
             );
         }
 
-        // Remove any empty group view that may have been created by
-        // addChoiceViewGroupedBy()
-        foreach ($preferredViews as $key => $view) {
```

Comment:
I don't think this change is backwards compatible. What if I create the following choices (which may come from a dynamic source):

```
$builder->add('field', ChoiceType::class, [
    'choices' => [
        'MyEmptyGroup' => [],
        'NonEmptyGroup' => [
            'A' => 'a_option',
            'B' => 'b_option',
        ],
    ],
]);
```

Before, `MyEmptGroup` would be removed. After, it wouldn't.

It's probably worth adding a test for this case.

Reply:
@webmozart , it is never created in this case, see https://github.com/symfony/symfony/blob/95d46c25f798519b51079d8360f85d615684b9f7/src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php#L179